packages/flutter_tools/gradle/build.gradle.kts

```diff
@@ -30,5 +30,5 @@ dependencies {
     //  * ndkVersion in FlutterExtension in packages/flutter_tools/gradle/src/main/flutter.groovy
     //  * AGP version constants in packages/flutter_tools/lib/src/android/gradle_utils.dart
     //  * AGP version in buildscript block in packages/flutter_tools/gradle/src/main/flutter.groovy
-    compileOnly("com.android.tools.build:gradle:7.3.0")
+    compileOnly("com.android.tools.build:gradle:7.4.2")
```

Comment:
If you're targeting Java 8, you might get into conflicting versions between the project and AGP artifacts. See [here](https://github.com/TWiStErRob/net.twisterrob.gradle/blob/0.16/gradle/plugins/src/main/kotlin/net/twisterrob/gradle/build/compilation/JavaCompatibilityPlugin.kt) for the latest variant of this problem, or [here](https://github.com/TWiStErRob/net.twisterrob.gradle/pull/364/files#diff-e2e837f5947404283a07c9a53beef24153ed15da97493c0dba0ac80238f7d7a6) for an older version that matches your tried upgrade.

Reply:
thanks, it'll for sure come in useful in a future PR when Gradle & AGP versions will be bumped.

btw the first links 404's

Reply:
Yeah, defo separate PR, it's a tricky one; fixed link, sorry.